lib/date.nit

```diff
@@ -0,0 +1,215 @@
+# limitations under the License.
+#
+# The date and time behavior is splited in 2 classes:
+# * `Date` handles the year, month and day part of the date.
+# * `Time` handles the time in hour, minute and second.
+# 
+# These are extended by DateTime that merges the two concepts in one class.
+#
+#	###################
+#	# Some operations #
+#	###################
+#
+#	#DateTime:
+#	var date_1  = new DateTime(2014, 02, 09, 13, 50, 0)
+#	var date_2  = new DateTime(2014, 02, 09, 13, 51, 0)
+#	
+#	assert date_1 > date_2
+# 	
+#	#Date:
+#	var date_1 = new Date(2014, 01, 12)
+#	var date_2 = new Date(2014, 01, 30)
+#	
+#	assert date_1.diff_month(date_2)
+#
+#	#Time:
+#	var time_1 = new Time.now
+#	var time_2 = new Time(14,25,30)
+#	
+#	assert time_1 == time_2
+#
+module date
+
+# The class "Time" allow the manipulation only on the time data
+#
+# 		var now = new Time.now ==> Set the current time
+# 		var time = new Time(14,30,00) ==> (Hours, Minutes, Seconds)
+class Time
+	super Comparable
+
+	# Hours as an integer
+	var hour: Int
+
+	# Minutes as an integer
+	var minute: Int
+
+	# Seconds as an integer
+	var second: Int
+
+	# This will initialize the current time
+	init now do
+		var tm = new Tm.localtime
+		hour = tm.hour
+		minute = tm.min
+		second = tm.sec
+	end
+
+	redef fun ==(d)  
+	do
+		return d isa Time and time_eq(d)
+	end
+
+	# This check if the time is eq or not
+	fun time_eq(d: Time): Bool
+	do
+		if (hour * 3600 + minute * 60 + second) != (d.hour * 3600 + d.minute * 60 + d.second) then
+			return false
+		end
+		return true
+	end
+end
+
+# The class "DateTime" allow the manipulation of date and time data
+#
+# 		var today = new DateTime.today ==> set the current date and time
+#		var date_time = DateTime(2014, 02, 09, 13, 50, 00) ==> (Year, Month, Day, Hours, Minutes, Seconds)
+class DateTime
+	super Date
+	super Time
+	autoinit year, month, day, hour, minute, second
+
+	redef init today do
+		super
+		var tm = new Tm.localtime
+		hour = tm.hour
+		minute = tm.min
+		second = tm.sec
+	end
+
+	redef fun ==(d)  
+	do
+		return d isa DateTime and diff_days(d) == 0 and time_eq(d)
+	end
+
+	redef fun to_s do
+		return super + " {hour.to_s}:{minute.to_s}:{second.to_s}{time_zone}"
+	end
+end
+
+# The class "Date" allow the manipulation only on the date data
+#
+#		var today = new Date ==> set the current date
+#		var date = new Date(2014, 03, 04) ==> (Year, Month, Day)
+class Date
+	super Comparable
+
+	# Year as an integer ex: 1989
+	var year: Int
+
+	# Month as an integer ex: "3" for March
+	var month: Int
+
+	# Day as an integer 
+	var day: Int
+
+	# UTC time zone
+	var time_zone = "Z"
+
+	# This will initialize the current date
+	init today do
+		var tm = new Tm.localtime
+		year = 1900 + tm.year
+		month = tm.mon + 1
+		day = tm.mday
+	end
+
+	# Date expressed according to ISO 8601
+	redef fun to_s 
+	do
+		var y = year.to_s
+		var mth = month.to_s
+		var d = day.to_s
+
+		return "{y}-{mth}-{d}"
```

Comment:
Superstrings always apply a `to_s`, you can change that by "{year}-{month}-{day}"